python_backend/api/v1/endpoints/quality.py

```
@@ -180,83 +217,71 @@ async def save_user_feedback(
+                'totalApplied': len(feedback_items)
+            },
             originalLength=len(request.original_text),
-            finalLength=len(request.original_text),
-            message=f"통합본 생성 실패: {str(e)}"
+            finalLength=len(final_text),
+            message="AI 기반 최종 통합본이 성공적으로 생성되었습니다."
         )
+        
+    except Exception as e:
+        logger.error(f"최종 통합본 생성 중 오류: {e}", exc_info=True)
+        raise HTTPException(status_code=500, detail=f"최종 통합본 생성 중 서버 오류 발생: {e}")
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)